### PR TITLE
feat: define region server related messages

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

define a new service for datanode (region server). Compared to the existing `GreptimeDatabase`, the new `RegionServer` is considered internal.

The planned usage of this service is to handle all requests to region, except the query request, which will be handled in flight.

## Checklist

- [ ]  I have written the necessary comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: proto/greptime/v1/region/server.proto [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+message InsertRequest { repeated Row rows = 1; }
+
+message DeleteRequest { repeated Row rows = 1; }
+
+message CreateRequest {
+  // Region engine name
+  string engine = 1;
+  // Columns in this region.
+  // todo:  pub column_metadatas : Vec<ColumnMetadata>,
```
Comment (evenyag) [inline proto/greptime/v1/region/server.proto:60]: Where is the column?

File: proto/greptime/v1/region/server.proto [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+  // Columns in the primary key.
+  repeated ColumnId primary_key = 3;
+  // Create region if not exists.
+  bool create_if_not_exists = 4;
+  // Directory for region's data home. Usually is composed by catalog and
```
Comment (evenyag) [inline proto/greptime/v1/region/server.proto:66]: and what?
```suggestion
  // Directory for region's data home. Usually is composed by catalog and ?????
```

File: proto/greptime/v1/region/server.proto [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package greptime.v1.region;
+
+option java_package = "io.greptime.v1";
```
Comment (evenyag) [inline proto/greptime/v1/region/server.proto:19]:
```suggestion
option java_package = "io.greptime.v1.region";
```

File: proto/greptime/v1/region/server.proto [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+option java_outer_classname = "Server";
+option go_package = "github.com/GreptimeTeam/greptime-proto/go/greptime/v1";
```
Comment (evenyag) [inline proto/greptime/v1/region/server.proto:21]:
```suggestion
option go_package = "github.com/GreptimeTeam/greptime-proto/go/greptime/v1/region";
```

File: proto/greptime/v1/common.proto [hunk 5]
```diff
@@ -69,13 +67,18 @@ enum SemanticType {
...
 
+// TODO: deprecate this, and use the `ColumnDef` in region_server.proto instead
 message ColumnDef {
   string name = 1;
   ColumnDataType datatype = 2;
   bool is_nullable = 3;
   bytes default_constraint = 4;
 }
 
+message ColumnId { uint32 column_id = 1; }
```
Comment (evenyag) [inline proto/greptime/v1/common.proto:78]: If column id isn't optional, maybe we don't need to define it as a new message.

File: proto/greptime/v1/region/server.proto (hunk 4 above)
Comment (fengjiachun) [inline proto/greptime/v1/region/server.proto:21]: Do we still need them(java/go)?

Comment (waynexia) [reply proto/greptime/v1/region/server.proto:21]: At present only rust is needed. So remove other languages?

**Review by evenyag — APPROVED** [review]

File: proto/greptime/v1/region/server.proto (hunk 4 above)
Comment (fengjiachun) [reply proto/greptime/v1/region/server.proto]: I think we can remove other languages

**Review by fengjiachun — APPROVED** [review]
LGTM

File: proto/greptime/v1/region/server.proto (hunk 4 above)
Comment (waynexia) [reply proto/greptime/v1/region/server.proto]: I find only go need to generate specifically, other are auto included. I've reverted the go script

**Pull request merged**